src/ethereum/arrow_glacier/transactions.py

```diff
@@ -0,0 +1,121 @@
+from dataclasses import dataclass
+from typing import Tuple, Union
+
+from .. import rlp
+from ..base_types import (
+    U64,
+    U256,
+    Bytes,
+    Bytes0,
+    Bytes8,
+    Bytes20,
+    Bytes32,
+    Bytes256,
+    Uint,
+    slotted_freezable,
+)
+from ..crypto.hash import Hash32
+from ..exceptions import InvalidBlock
+
+Address = Bytes20
+Root = Hash32

```

Comment:
These definitions don't really make sense in the `transactions.py` module. I assume you moved them because of a circular dependency problem? My next few comments will be a proposed solution.